R/derive_var_dthcaus.R

```diff
@@ -153,17 +153,29 @@ derive_var_dthcaus <- function(dataset,
   for (ii in seq_along(sources)) {
     source_dataset_name <- sources[[ii]]$dataset_name
     source_dataset <- source_datasets[[source_dataset_name]]
-    if (!quo_is_null(sources[[ii]]$filter)) {
+    if (!is.null(sources[[ii]]$date_imputation) & !quo_is_null(sources[[ii]]$filter)) {
       add_data[[ii]] <- source_dataset %>%
-        filter(!!sources[[ii]]$filter)
+        filter(!!sources[[ii]]$filter) %>%
+        mutate(temp_date = impute_dtc(dtc = !!sources[[ii]]$date,
+                                      date_imputation = !!sources[[ii]]$date_imputation))
+    } else if (!is.null(sources[[ii]]$date_imputation)) {
+      add_data[[ii]] <- source_dataset %>%
+        mutate(temp_date = impute_dtc(dtc = !!sources[[ii]]$date,
+                                      date_imputation = !!sources[[ii]]$date_imputation))
+    } else if (!quo_is_null(sources[[ii]]$filter)) {
+      add_data[[ii]] <- source_dataset %>%
+        filter(!!sources[[ii]]$filter) %>%
+        mutate(temp_date = !!sources[[ii]]$date)
     } else {
-      add_data[[ii]] <- source_dataset
+      add_data[[ii]] <- source_dataset %>%
+        mutate(temp_date = !!sources[[ii]]$date)
     }
+    add_data[[ii]][6] <- add_data[[ii]][7]
```

Comment:
Please do not access columns by number. It is hard to understand and unlikely that it works in general.

I think we do not need this line at all if you use `temp_date` below.

Reply:
I've removed this line - the result is that the duplicate error message reads: 
Dataset contains duplicate records with respect to `STUDYID`, `USUBJID` and `temp_date`
Is this okay or is there another way to update this?

Reply:
I see. Maybe we should rename `temp_date` to `temp_imputed_date`. I think then it is clearer.

Reply:
I would call `impute_dtc()` even if `date_imputation` is `NULL`. Then partial dates are set to `NA`. This is consistent with other functions and in line with the documentation.